document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -478,6 +505,10 @@ public static String prepareParsedNonCompositeFilter(
         throw new UnsupportedOperationException(UNSUPPORTED_QUERY_OPERATION);
     }
 
+    if (isInOP) {
+      return filterString.toString();
+    }
```

Comment:
Could we simply return from the switch-case directly and avoid the boolean flag?